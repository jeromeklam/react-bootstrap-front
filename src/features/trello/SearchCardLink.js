import React from 'react';
import PropTypes from 'prop-types';

export default function SearchCardLink(props) {
  return (
    <button className="btn btn-primary" onClick={props.onClick}>
      {props.t('Click to search a card')}
    </button>
  );
}

SearchCardLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
};

SearchCardLink.defaultProps = {};
