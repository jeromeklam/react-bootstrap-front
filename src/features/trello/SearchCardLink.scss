@import '../../styles/mixins';

.trello-search-card-link {

}
