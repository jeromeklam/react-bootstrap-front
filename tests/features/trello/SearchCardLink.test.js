import React from 'react';
import renderer from 'react-test-renderer';
import SearchCardLink from '../../../src/features/trello/SearchCardLink';

const callback = jest.fn();

it('render with className and name', () => {
  // First render
  const component = renderer.create(
    <SearchCardLink className="btn bg-secondary" t={callback} onClick={callback}/>
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});