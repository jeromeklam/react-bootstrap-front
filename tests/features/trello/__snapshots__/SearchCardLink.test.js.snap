// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with className and name 1`] = `
<button
  className="btn btn-primary"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [
          "Click to search a card",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
/>
`;
